eq-author/src/App/page/Logic/index.js

```diff
@@ -67,36 +69,80 @@ const LogicLink = styled(NavLink)`
   }
 `;
 
+const Badge = styled.span`
+  border-radius: 0.7em;
+  border: 1px solid ${colors.white};
+  background-color: ${colors.red};
+  color: white;
+  padding: 0.1em 0.3em;
+  font-weight: normal;
+  z-index: 2;
+  margin-left: auto;
+  line-height: 1;
+  font-size: 0.9rem;
+  pointer-events: none;
+  width: 20px;
+  height: 20px;
+`;
+
 export class UnwrappedLogicPage extends React.Component {
   static propTypes = {
     children: PropTypes.node.isRequired,
-    page: CustomPropTypes.page,
-    match: PropTypes.shape({
-      params: PropTypes.shape({
-        questionnaireId: PropTypes.string.isRequired,
-      }).isRequired,
-    }).isRequired,
+    data: PropTypes.shape({
+      page: CustomPropTypes.page,
+    }),
+    loading: PropTypes.bool.isRequired,
+    error: PropTypes.object, // eslint-disable-line
   };
 
   renderContent() {
-    const { children } = this.props;
+    const { children, data, loading, error } = this.props;
+    const page = get(data, "page", null);
+
+    if (loading) {
+      return <Loading height="20em">Loading routing</Loading>;
+    }
+
+    if (error || !page) {
+      return <Error>Something went wrong</Error>;
+    }
+
+    const TABS = [
+      {
+        key: `routing`,
+        label: "Routing logic",
+      },
+      {
+        key: `skip`,
+        label: "Skip logic",
+      },
+    ];
 
     return (
       <LogicMainCanvas>
         <Grid>
           <Column gutters={false} cols={2.5}>
             <MenuTitle>Select your logic</MenuTitle>
             <StyledUl>
-              <li>
-                <LogicLink exact to={`routing`} activeClassName="active">
-                  Routing logic
-                </LogicLink>
-              </li>
-              <li>
-                <LogicLink exact to={`skip`} activeClassName="active" replace>
-                  Skip logic
-                </LogicLink>
-              </li>
+              {TABS.map(({ key, label }) => {
+                const errors = filter(page.validationErrorInfo.errors, error =>
+                  error.id.includes(key)
+                );
```

Comment:
If instead you check page has loaded correctly here, the page remains intact on tab changes and still safeguards when the component is loading.

Something like this
```suggestion
           let errors;
                if (page) {
                  errors = filter(page.validationErrorInfo.errors, error =>
                    error.id.includes(key)
                  );
                }
```

Reply:
You could just put the loading test on the badge icon only
```
{!loading &&
  errors !== undefined &&
  errors !== null &&
  errors.length > 0 ? (
  <Badge data-test="badge-withCount">
  {errors.length}
   </Badge>
  ) : null}
```
and then delete the loading test on line 102.
This would keep the static part of the labels there then.

Reply:
This works well. I don't why. but it works. lol